core/src/mender-client.c

```diff
@@ -1143,6 +1150,25 @@ mender_client_update_work_function(void) {
                 if (NULL != mender_update_module->callbacks[update_state]) {
                     ret = mender_update_module->callbacks[update_state](update_state, (mender_update_state_data_t)NULL);
                 }
+
+                if (MENDER_OK != ret) {
+                    /* If the rollback verify reboot fails, we will retry the
+                     * rollback reboot CONFIG_MENDER_ROLLBACK_REBOOT_LOOP_RETRIES times
+                     * before going to the failure state.
+                     */
+                    size_t retry_rollback_reboot_counter;
+                    mender_deployment_data_retry_rollback_reboot_counter(mender_client_deployment_data, &retry_rollback_reboot_counter);
+                    if (retry_rollback_reboot_counter <= CONFIG_MENDER_ROLLBACK_REBOOT_LOOP_RETRIES) {
+                        mender_log_info("Rollback verify reboot failed. Retry rollback reboot %zu/%d",
+                                        retry_rollback_reboot_counter,
+                                        CONFIG_MENDER_ROLLBACK_REBOOT_LOOP_RETRIES);
+                        update_state = MENDER_UPDATE_STATE_ROLLBACK_REBOOT;
+                        mender_log_debug("Entering state %s", update_state_str[update_state]);
+                        set_and_store_state(update_state);
+                        continue;
```

Comment:
We will need some wait between retries. I don't think we have anything else in the code _waiting_, though so I don't know how to handle it.

Reply:
We have a `k_sleep` in mender-net.c, should I add a sleep? Do you have a suggestion or an interval in mind?

Reply:
As discussed in person,

I was wrong here. This is looping through two separate states, therefore we don't need to sleep between them...